src/core/mac/mac.cpp

```diff
@@ -269,7 +269,11 @@ ThreadError Mac::SetNetworkName(const char *aNetworkName)
     VerifyOrExit(strlen(aNetworkName) <= OT_NETWORK_NAME_MAX_SIZE, error = kThreadError_InvalidArgs);
 
     memset(&mNetworkName, 0, sizeof(mNetworkName));
+#ifdef _WIN32
+    strcpy_s(mNetworkName.m8, sizeof(mNetworkName), aNetworkName);
+#else
```

Comment:
This could likely be spun off into it's own pull request, considering that it affects all platforms.